File: src/main/java/com/ddd/trainings/domain/TrainerId.java
package com.ddd.trainings.domain;

import lombok.EqualsAndHashCode;
import lombok.Getter;

@EqualsAndHashCode(of = "value")
public class TrainerId {

  @Getter
  private final String value;

  private TrainerId(String value) {
    this.value = value;
  }

  public static TrainerId from(String value) {
    return new TrainerId(value);
  }
}

File: src/Ddd/Trainings/Idea/Domain/TrainerId.cs
namespace Ddd.Trainings.Idea.Domain;

public class TrainerId(string value)
{
    public string Value { get; } = value;

    public static TrainerId From(string value) => new(value);

    public override bool Equals(object? obj)
    {
        if (obj is not TrainerId other)
            return false;
        return Value == other.Value;
    }

    public override int GetHashCode() => Value.GetHashCode();
} 